File: app/src/main/java/com/example/jualhp/data/model/Product.kt
package com.example.jualhp.data.model

// Data class untuk response dari API terkait kategori produk.
// Setiap kategori memiliki ID, nama kategori, dan daftar produk di dalamnya.
data class CategoryProductResponse(
    val categoryId: String, // ID dari kategori
    val categoryName: String, // Nama kategori
    val products: List<Product> // Daftar produk dalam kategori
)

// Data class untuk merepresentasikan data produk.
// Setiap produk memiliki atribut-atribut seperti ID, judul, deskripsi, harga, rating, dan URL gambar.
data class Product(
    val id: Int, // ID unik dari produk
    val title: String, // Judul atau nama produk
    val description: String, // Deskripsi produk
    val price: Int, // Harga produk dalam bentuk integer
    val rating: Float, // Rating produk dalam skala float
    val image: String // URL atau path untuk gambar produk
)

File: lib/models/product.dart
class Product {

  final int id;
  final String title;
  final String description;
  final int price;
  final double rating;
  final String image;

  // untuk membuat objek Product dengan data yang diperlukan
  Product({
    required this.id,
    required this.title,
    required this.description,
    required this.price,
    required this.rating,
    required this.image,
  });

  // mengubah data JSON menjadi objek Product
  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],  // ID produk diambil dari JSON
      title: json['title'],  // Judul produk diambil dari JSON
      description: json['description'],  // Deskripsi produk diambil dari JSON
      price: json['price'],  // Harga produk diambil dari JSON
      rating: json['rating'].toDouble(),  // Rating produk diambil dari JSON, diubah menjadi double
      image: json['image'],  // URL gambar produk diambil dari JSON
    );
  }
}
